src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/CompletionHandler.cs

```diff
@@ -432,6 +435,61 @@ private CompletionList IncludeCSharpKeywords(CompletionList completionList)
             return completionList;
         }
 
+        internal CompletionList TranslateTextEdits(
+            Position hostDocumentPosition,
+            Position projectedPosition,
+            TextExtent? wordExtent,
+            CompletionList completionList)
+        {
+            var wordRange = wordExtent == null ? null : wordExtent.Value.Span.AsRange();
+            var newItems = completionList.Items.Select(item => TranslateTextEdits(hostDocumentPosition, projectedPosition, wordRange, item)).ToArray();
+            completionList.Items = newItems;
+
+            return completionList;
+
+            static CompletionItem TranslateTextEdits(Position hostDocumentPosition, Position projectedPosition, Range wordRange, CompletionItem item)
+            {
+                if (item.TextEdit != null)
+                {
+                    var offset = projectedPosition.Character - hostDocumentPosition.Character;
```

Comment:
We should add some large comments about the reasoning behind the "offset" approach